cpu/riscv_common/include/pmp.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2023 Bennet Blischke <bennet.blischke@haw-hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup         cpu_riscv_common
+ * @{
+ *
+ * @file
+ * @brief           RISC-V PMP configuration options
+ *
+ * RISCV implementations using this peripheral must define the `NUM_PMP_ENTRIES`
+ * `NUM_PMP_ENTRIES` must be 16 or 64.
+ *
+ * @author          Bennet Blischke
+ */
+
+#ifndef PMP_H
+#define PMP_H
+
+#include <stdint.h>
+#include "cpu_conf.h"
+#include "vendor/riscv_csr.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define CSR_PMPCFG0 0x3a0
+#define CSR_PMPCFG1 0x3a1
+#define CSR_PMPCFG2 0x3a2
+#define CSR_PMPCFG3 0x3a3
+#define CSR_PMPADDR0 0x3b0
+#define CSR_PMPADDR1 0x3b1
+#define CSR_PMPADDR2 0x3b2
+#define CSR_PMPADDR3 0x3b3
+#define CSR_PMPADDR4 0x3b4
+#define CSR_PMPADDR5 0x3b5
```

Comment:
This block could be moved to `vendor/riscv_csr.h`. Should I?

Reply:
sure, since that's where the other `CSR_` defines live already

Reply:
But the file says: `/* Automatically generated by parse-opcodes */` - should I edit a file thats being generated?

Reply:
but the rest of that file doesn't appear to be generated - unless @bergzand objects, I'd say just it it there.

Reply:
I'm fine with adding it there. If it is indeed generated, it is a bit sub-optimal as these CSRs are not in there already.

Reply:
Moved it :woman_shrugging:. Also got rid of the single address defines and went for "this is the base reg" as my code does that already anyways.